Examples/Io/Csv/src/CsvTrackingGeometryWriter.cpp

```diff
@@ -76,76 +78,67 @@ void writeSurface(SurfaceWriter& writer, const Acts::Surface& surface,
   data.rot_zv = transform(2, 1);
   data.rot_zw = transform(2, 2);
 
-  // module thickness
-  if (surface.associatedDetectorElement()) {
-    const auto* detElement =
-        dynamic_cast<const Acts::IdentifiedDetectorElement*>(
-            surface.associatedDetectorElement());
-    if (detElement) {
-      data.module_t = detElement->thickness() / Acts::UnitConstants::mm;
-    }
-  }
+  std::array<float*, 7> dataBoundParameters = {
+      &data.p0, &data.p1, &data.p2, &data.p3, &data.p4, &data.p5, &data.p6};
```

Comment:
I'm not sure I understand why this needs to be an array of `float` pointers. Can you fill me in?

Reply:
This my understanding: you can not have an array of elements in the struct but you want to fill them in from a variable container. By creating a container of pointers you can keep the separate variables in the struct and iterate over them here.

Reply:
yup.